src/cls/rgw/cls_rgw_client.h

```diff
@@ -27,16 +28,35 @@ struct BucketIndexAioArg : public RefCountedObject {
 };
 
 /*
- * This class manages AIO completions. This class is not completely thread-safe,
- * methods like *get_next* is not thread-safe and is expected to be called from
- * within one thread.
+ * This class manages AIO completions. This class is not completely
+ * thread-safe, methods like *get_next_request_id* is not thread-safe
+ * and is expected to be called from within one thread.
  */
 class BucketIndexAioManager {
+public:
+
+  // allows us to reaccess the shard id and shard's oid during and
+  // after the asynchronous call is made
+  struct RequestObj {
+    const int shard_id;
+    const std::string oid;
+
+    RequestObj(const int _shard_id, const std::string& _oid) :
+      shard_id(_shard_id), oid(_oid)
+    {/* empty */}
+
+    RequestObj(const RequestObj&) = default;
```

Comment:
if you had to re-enable the copy ctor, you should do the move constructor too

Reply:
Removed as not needed.